context-propagation/src/main/java/io/micrometer/context/ThreadLocalAccessor.java

```diff
@@ -53,9 +65,28 @@
     void setValue(V value);
 
     /**
-     * Remove the {@link ThreadLocal} value.
+     * Called instead of {@link #setValue(Object)} in order to remove the current
+     * {@link ThreadLocal} value at the start of a
+     * {@link io.micrometer.context.ContextSnapshot.Scope}.
+     *
+     * @since 1.0.3
      */
-    void reset();
+    default void setValue() {
+        reset();
+    }
+
+    /**
+     * Remove the {@link ThreadLocal} value when setting {@link ThreadLocal} values in
+     * case of missing mapping for a {@link #key()} from a {@link ContextSnapshot}, or a
+     * Context object (operated upon by {@link ContextAccessor}).
+     * @deprecated To be replaced by calls to {@link #setValue()}, which needs to be
+     * implemented when this implementation is removed.
+     */
+    @Deprecated
+    default void reset() {
+        throw new IllegalStateException(this.getClass().getName() + "#reset() should "
+                + "not be called. Please implement #setValue() method when removing the " + "#reset() implementation.");
+    }
```

Comment:
I think a typical implementation would only need one method to clear a `ThreadLocal`. I see two of the test implementations (besides the Observation one) implement `setValue()` and `restore()` delegates to it. From the perspective of a typical implementation it makes more sense to implement `reset()` while `setValue()` and `restore()` are not required and simply delegate to `reset()`. That makes me think we should keep `reset()` rather than deprecating it.